WalletWasabi.Fluent/Models/Transactions/PrivacySuggestionsModel.cs

```diff
@@ -293,7 +293,18 @@ await foreach (var suggestion in suggestions)
 			result.Add(suggestion);
 		}
 
-		return result;
+		// If there's one or zero suggestion, or one is Less, the other is More, simply return the results.
+		// If both is Less/More, only return the one with smaller difference.
+		if (result.Count <= 1 || result[0].IsLess != result[1].IsLess)
+		{
+			return result;
+		}
+		else
+		{
+			int indexOfBigger = Math.Abs(result[0].Difference) < Math.Abs(result[1].Difference) ? 1 : 0;
+			result.RemoveAt(indexOfBigger);
+			return result;
+		}
```

Comment:
Taking a look at the `foreach` at L274, I can see there are already similar "fixes", so I think it would make sense to add this next to them too.

Somehow like this:
```c#
if (changeAvoidanceSuggestions.FirstOrDefault(x => x.IsLess == suggestion.IsLess && x.IsMore == suggestion.IsMore) is { } existingSuggestion &&
    Math.Abs(suggestion.Difference) < Math.Abs(existingSuggestion.Difference))
{
	result.Remove(existingSuggestion);
}
```

I did not test it, but it should work when there are two less or two more options.

WDYT?

Reply:
> it would make sense to add this next to them too. 

Agreed.

A code's idea is good, but it doesn't work. It removes `existingSuggestion` if it's the bigger one, but if not, the leftover code will just add `suggestion` to `result` and return both.
Tweaked this in https://github.com/zkSNACKs/WalletWasabi/pull/11789/commits/09d22eac28caf8b73806eddcfa55dd7aa0214815, now it works.

Reply:
Nice